### PR TITLE
My VA: remove dashboard v1 components

### PR DESCRIPTION
**Pull request by dellerbie**
Removes legacy code from the My VA dashboard application, 

## Summary

 - Remove old components as outlined above
 - Rename v2 components, removing "v2"
 - Update tests as needed

## Related issue(s)

- [62899](https://github.com/department-of-veterans-affairs/va.gov-team/issues/62899)
- [60360](https://github.com/department-of-veterans-affairs/va.gov-team/issues/60360)

## Testing done

- Updated unit and e2e tests
- localhost visual verification

## What areas of the site does it impact?
My VA Dashboard

## Acceptance criteria
- [ ] Old components and associated tests are removed

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/personalization/dashboard/components/health-care/Appointments.jsx [hunk 1]
```diff
@@ -22,29 +26,18 @@ export const Appointments = ({ appointments, hasError }) => {
...
-          <p>
-            <CTALink
-              text="Schedule and manage your appointments"
-              href="/health-care/schedule-view-va-appointments/appointments"
-            />
-          </p>
-        </va-alert>
-      </div>
-    );
-  }
```
Comment (adamwhitlock1) [inline src/applications/personalization/dashboard/components/health-care/Appointments.jsx:44]: Is the error handling being done somewhere else, if so then we should be able to remove this prop from the component/prop types. If not, then we should probably still handle it here.

Comment (adamwhitlock1) [reply src/applications/personalization/dashboard/components/health-care/Appointments.jsx:44]: Nevermind I see the HealthcareError component

**Review by adamwhitlock1 — APPROVED** [review]
Wow that was a lot of work @dellerbie I had a few questions, but overall i dont see anything blocking. Locally everything looks good, and I think all the files and naming things are generally up to date. 

Do you think in the future you could make this PR a couple different PRs and migrate one part at a time, it might make things a bit more reviewable.

Regardless, great work on this, I know maintenance and deprecating things is not the most thrilling.

File: src/applications/personalization/dashboard/components/claims-and-appeals/Appeal.jsx [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import moment from 'moment';
 
 import {
```
Comment (allisonlu) [inline src/applications/personalization/dashboard/components/claims-and-appeals/Appeal.jsx:5]: similar to Adam's comment above. should we rename `appeals-v2-helpers.jsx` to just `appeals-helpers.jsx?`

File: src/applications/personalization/dashboard/components/benefit-payments/PaymentsCard.jsx [hunk 3]
```diff
@@ -4,17 +4,17 @@ import { format } from 'date-fns';
 import CTALink from '../CTALink';
 import recordEvent from '~/platform/monitoring/record-event';
 
-export const PaymentsV2 = ({ lastPayment }) => {
+export const Payments = ({ lastPayment }) => {
```
Comment (allisonlu) [inline src/applications/personalization/dashboard/components/benefit-payments/PaymentsCard.jsx:7]: should we rename this to `PaymentsCard`?

**Comment by allisonlu** [comment]
i also don't have launch-blocking comments, and it all looks great on the UI side! thanks for taking this on 🦅 👀 

just wanted to note that there are some files with `-v2` in their names in /dashboard/reducers/ and /utils/

**Review by allisonlu — APPROVED** [review]

**Pull request merged**